client/signup/config/flows.js

```diff
@@ -194,13 +194,15 @@ function getDIFMSiteContentCollectionDestination( { siteSlug } ) {
 	return `/home/${ siteSlug }`;
 }
 
-function getHostingFlowDestination( { siteId } ) {
-	return addQueryArgs(
-		{
-			'new-site': siteId,
-			'hosting-flow': true,
-		},
-		'/sites'
+function getHostingFlowDestination() {
+	return (
+		getQueryArgs().redirect_to ??
```

Comment:
We recently had issues reported with `redirect_to` params not being validated propertly. Read more at p7H4VZ-4sH-p2.

We have a `/me/validate-redirect` endpoint which can authoritatively confirm whether a redirect is safe. However introducing an async network call here would be problematic. We could also use a regex like #79189 to confirm the redirect is going to the `new-hosted-site` flow.

Another, perhaps even more straightforward solution, would be not to have a generic `?redirect_to` param, but a `?redirect_to_new_hosted_site_flow` query param that always redirects to the well known `/setup/new-hosted-flow` URL.

Reply:
Or maybe not redirecting at all like you suggest in your PR description and just going to the forking step?